CppCoreGuidelines.md

```diff
@@ -3528,13 +3528,31 @@ There is not a choice when a set of functions are used to do a semantically equi
 
     ???
 
-### <a name="Rf-reference-capture"></a>F.52: Prefer capturing by reference in lambdas that will be used locally, including passed to algorithms
+### <a name="Rf-reference-capture"></a>F.52: When a lambda stays in local scope, prefer capturing by reference, including passed to algorithms
 
 ##### Reason
 
-For efficiency and correctness, you nearly always want to capture by reference when using the lambda locally. This includes when writing or calling parallel algorithms that are local because they join before returning.
+When a lambda is only used locally, capturing by reference is always more efficient.  This includes calling parallel algorithms because they join before returning.
```

Comment:
For any case where you want to mutate the data being passed, then missing the capture-by-reference would be incorrect. Since the code would still compile, I suspect that's what 'correctness' is after here.

Simple example would be using `std::for_each` to take a vector of IP addresses and append a port to them.

I think the interesting question then becomes: is it not the case that an equal and opposite could be true? As in - are there cases when mutating the reference would be incorrect, and passing by value is better?

Suppose you want to create a new vector containing IP addresses with ports appended, and leave the input array unchanged:

```
int main()
{
  std::vector<std::string> blob = {"::", "192.168.0.1", "127.0.0.1"};
  std::vector<std::string> out(3);
  std::transform(blob.begin(), blob.end(), out.begin(), [&] (auto const x) { return x + "::80"; } );
 
  std::for_each(out.begin(), out.end(), [&] (auto const x) { std::cout << x << '\n'; });
}
```

In this case, passing by reference is more efficient, and still correct (the input vector is `const`).

Can you think of an example (of good code) where *not* passing by reference is more correct?